src/completions/Bindings.ts

```diff
@@ -0,0 +1,152 @@
+import * as Completions from "@src/completions"
+import * as config from "@src/lib/config"
+
+class BindingsCompletionOption extends Completions.CompletionOptionHTML
+    implements Completions.CompletionOptionFuse {
+    public fuseKeys = []
+
+    constructor(
+        public value: string,
+        binding: { name: string; value: string; mode: string },
+    ) {
+        super()
+        this.html = html`<tr class="BindingsCompletionOption option">
+                <td class="name">${binding.name}</td>
+                <td class="content">${binding.value}</td>
+                <td class="type">${binding.mode}</td>
+            </tr>`
+    }
+}
+
+export class BindingsCompletionSource extends Completions.CompletionSourceFuse {
+    public options: BindingsCompletionOption[]
+
+    constructor(private _parent) {
+        super(
+            ["bind", "unbind", "bindurl", "unbindurl", "reset", "reseturl"],
+            "BindingsCompletionSource",
+            "Bindings",
+        )
+
+        this._parent.appendChild(this.node)
+    }
+
+    public async filter(exstr: string) {
+        this.lastExstr = exstr
+        let options = ""
+        let [prefix, query] = this.splitOnPrefix(exstr)
+        const args = query ?  query.split(/\s+/) : []
+        let configName: string = "nmaps"
+        let modeName = "normal"
+        let urlPattern: string  = null
+
+        // Hide self and stop if prefixes don't match
+        if (prefix) {
+            // Show self if prefix and currently hidden
+            if (this.state === "hidden") {
+                this.state = "normal"
+            }
+        } else {
+            this.state = "hidden"
+            return
+        }
+
+        // url pattern is mandatory: bindurl, unbindurl, reseturl
+        if (prefix.trim().endsWith("url")) {
+            urlPattern = args.length > 0 ? args.shift() : ""
+            options += urlPattern ? urlPattern + " " : ""
+
+            if (args.length === 0) {
+                const patterns = config.get("subconfigs")
+                this.options = Object.keys(patterns)
+                    .filter(pattern => pattern.startsWith(urlPattern))
+                    .sort()
+                    .map(pattern => {
+                        return new BindingsCompletionOption(
+                            pattern, {
+                            name: pattern,
+                            value: "",
+                            mode: "URL Pattern",
+                        })
+                    })
+
+                return this.updateChain()
+            }
+        }
+
+        const mode2maps = new Map([
```

Comment:
Could this be imported from somewhere else? I'd rather have a single source of truth for it :)

Reply:
It's copied from excmds.ts where the comment said it might be part of rewrite of some bigger work(?). Put it in a `lib/binding.ts`?

https://github.com/tridactyl/tridactyl/blob/c6e1ed8c5ca9b4b8164c6ea7015b3bdd035962d8/src/excmds.ts#L3134-L3146

Reply:
A `lib/binding` would be great :)